speech_cleanup.qmd

```diff
@@ -0,0 +1,120 @@
+```{python}
+import pandas_gbq
+import os
+import pandas as pd
+import seaborn as sns
+import numpy as np
+import re
+```
+
+```{python}
+
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "token/gcp_token.json"
+
+df = pandas_gbq.read_gbq("""
+                    SELECT  *
+                    FROM `singapore-parliament-speeches.prod_mart.mart_speeches`
+                         WHERE member_name != ''
+                         AND member_name != 'Speaker'
+                         AND is_primary_question = false
+                    """)
+```
+
+```{python}
+[i for i in df.query('count_speeches_words<50').speech_text]
+
+```
+
+Some visuals, most speeches are within 2000 words with serious outliers
+
+```{python}
+sns.histplot(x = 'count_speeches_words', data = df)
+```
+
+What types of speeches?
+
+```{python}
+
+sns.histplot(y = 'topic_type_name', data = df)
+
+```
+
+What do short speeches look like?
+
+```{python}
+perc_10 = np.percentile(df.count_speeches_words, 10)
+[i for i in df.query(f'count_speeches_words<{perc_10}').speech_text]
+
+```
+
+What do different types of speeches look like?
+
+```{python}
+
+[i for i in df.query('topic_type_name.str.contains("Bill Introduced")').speech_text]
+
+[i for i in df.query('topic_type_name.str.contains("Correction by Written Statements")').speech_text]
+
+```
+
+Some patterns revealed
+1. Many procedural points - 'Yes please', 'Aye', 'last qns', etc. These are all very short and by speaker
+2. empty open brackets
+3. responses to speaker - 'thank you', etc.
```

Comment:
i've added a flag to identify speeches which are shorter than the 10th percentile (less than 4 words) and this can serve as a basis for filtering out short speeches.

something like:
```sql
count_words <= percentile_cont(count_words, 0.1) over ()
```

Reply:
See below